### PR TITLE
add some additional metrics, move verification for pgsync index into …

### PR DESCRIPTION
**Pull request by magdee3**
…should_skip_event

### CODE REVIEW

File: pgsync/sync.py [hunk 1]
```diff
@@ -1285,15 +1295,17 @@ def async_poll_db(self) -> None:
 
         while self.conn.notifies:
             notification: t.AnyStr = self.conn.notifies.pop(0)
+            self.count['notifications'] += 1
             if notification.channel == self.database:
                 payload = json.loads(notification.payload)
-                if self.index in payload["indices"]:
```
Comment (tolitius) [inline pgsync/sync.py:1290]: why is this removed?
seems to be needed and fix as per [fix](https://github.com/tolitius/pgsync/commit/de812696d9346e4a5a5a744fa50f9696daddcbef) in the upstream repo we don't yet have

File: pgsync/sync.py [hunk 2]
```diff
@@ -1421,8 +1433,11 @@ def _status(self, label: str) -> None:
             f"Xlog: [{self.count['xlog']:,}] => "
             f"Db: [{self.count['db']:,}] => "
             f"Redis: [{self.redis.qsize:,}] => "
-            f"{self.search_client.name}: [{self.search_client.doc_count:,}]"
-            f"...\n"
+            f"{self.search_client.name}: [{self.search_client.doc_count:,}] => "
+            f"skip-Xlog: [{self.count['skip_xlog']:,}] => "
+            f"skip-Redis: [{self.count['skip_redis']:,}] => "
+            f"Events: [{self.count['notifications']:,}]"
+            f"\n"
```
Comment (tolitius) [inline pgsync/sync.py:1439]: logs as:
```
Xlog: [3,185] => Db: [72,251] => Redis: [1] => OpenSearch: [63,190] => skip-Xlog: [2,867] => skip-Redis: [9,224] => Events: [81,475]
```

let's do these:
```
tx-log [3,185] => db-changes [72,251] => queued [1] => elastic [63,190]  |||  skipped: tx-log [2,867], queue [9,224]
```

File: pgsync/sync.py [hunk 3]
```diff
@@ -433,6 +433,12 @@ def should_skip_event(self, event: dict) -> bool:
 
         logger.debug(f"should_skip_event: validating event: {event}")
 
+        if event['indices'] is not None:
+            # event['indices'] set to None when pgsync parse replication slot
+            if self.index in [event['indices']]:
+                logger.debug(f"should_skip_event: skipping event, the evnet's index list names is not matche pgsync JSON schema name")
```
Comment (tolitius) [inline pgsync/sync.py]:
```python
logger.debug(f"should_skip_event: skipping event, none of the event's index names \"{event['indices']}\" matches pgsync JSON schema name")
```

File: pgsync/sync.py [hunk 4]
```diff
@@ -433,6 +433,12 @@ def should_skip_event(self, event: dict) -> bool:
 
         logger.debug(f"should_skip_event: validating event: {event}")
 
+        if event['indices'] is not None:
+            # event['indices'] set to None when pgsync parse replication slot
+            if self.index in [event['indices']]:
```
Comment (tolitius) [inline pgsync/sync.py:438]: skipping if self.index is **_IN_** the event['indices']?

**Review by tolitius — CHANGES_REQUESTED** [review]

File: pgsync/sync.py (hunk 1 above)
Comment (magdee3) [reply pgsync/sync.py:1290]: this condition moved into should_skip_event()

File: pgsync/sync.py (hunk 4 above)
Comment (magdee3) [reply pgsync/sync.py:438]: No, event['indices'] should be enumerate, can replace with enumerate(event['indices'])

File: pgsync/sync.py (hunk 2 above)
Comment (magdee3) [reply pgsync/sync.py:1439]: let's leave this as is for a while, or  I should refactor vector and grafana dashboards

Comment (tolitius) [reply pgsync/sync.py:1439]: k, let's do it in the follow up PRs
